parseany.go

```diff
@@ -605,131 +614,103 @@ iterRunes:
 		// 4/8/2014 22:05
 		// 04/08/2014 22:05
 		// 2014/4/8 22:05
 		// 2014/04/08 22:05
+		// 2014/04/08 08:05 PM
+
+		var t time.Time
+		var err error
+		var add time.Duration
+
+		if state == ST_DIGITSLASHWSCOLONAM {
+			datestr = strings.Replace(datestr, " AM", "", -1)
+		} else if state == ST_DIGITSLASHWSCOLONPM {
+			datestr = strings.Replace(datestr, " PM", "", -1)
+			add = time.Duration(12 * time.Hour)
+		}
+
 		if firstSlash == 4 {
-			if len(datestr) == len("2006/01/02 15:04") {
-				if t, err := time.Parse("2006/01/02 15:04", datestr); err == nil {
-					return t, nil
-				} else {
-					return time.Time{}, err
-				}
-			} else if len(datestr) == len("2006/01/2 15:04") {
-				if t, err := time.Parse("2006/01/2 15:04", datestr); err == nil {
-					return t, nil
-				} else {
-					if t, err := time.Parse("2006/1/02 15:04", datestr); err == nil {
-						return t, nil
-					} else {
-						return time.Time{}, err
-					}
-				}
-			} else {
-				if t, err := time.Parse("2006/1/2 15:04", datestr); err == nil {
-					return t, nil
-				} else {
-					return time.Time{}, err
+			switch len(datestr) {
+			case len("2006/01/02 15:04"):
+				t, err = time.Parse("2006/01/02 15:04", datestr)
+			case len("2006/01/2 15:04"):
+				if t, err = time.Parse("2006/01/2 15:04", datestr); err != nil {
+					t, err = time.Parse("2006/1/02 15:04", datestr)
 				}
+			default:
+				t, err = time.Parse("2006/1/2 15:04", datestr)
 			}
 		} else {
-			if len(datestr) == len("01/02/2006 15:04") {
-				if t, err := time.Parse("01/02/2006 15:04", datestr); err == nil {
-					return t, nil
-				} else {
-					return time.Time{}, err
-				}
-			} else if len(datestr) == len("01/2/2006 15:04") {
-				if t, err := time.Parse("01/2/2006 15:04", datestr); err == nil {
-					return t, nil
-				} else {
-					if t, err := time.Parse("1/02/2006 15:04", datestr); err == nil {
-						return t, nil
-					} else {
-						return time.Time{}, err
-					}
-				}
-			} else {
-				if t, err := time.Parse("1/2/2006 15:04", datestr); err == nil {
-					return t, nil
-				} else {
-					return time.Time{}, err
+			switch len(datestr) {
+			case len("01/02/2006 15:04"):
+				t, err = time.Parse("01/02/2006 15:04", datestr)
+			case len("01/2/2006 15:04"):
+				if t, err = time.Parse("01/2/2006 15:04", datestr); err != nil {
+					t, err = time.Parse("1/02/2006 15:04", datestr)
 				}
+			default:
+				t, err = time.Parse("1/2/2006 15:04", datestr)
 			}
 		}
-	case ST_DIGITSLASHWSCOLONCOLON, ST_DIGITSLASHWSCOLONCOLONAMPM: // starts digit then slash 02/ more digits/slashes then whitespace double colons
+		if err == nil {
+			return t.Add(add), nil
+		}
+		return time.Time{}, err
+
+	case ST_DIGITSLASHWSCOLONCOLON, ST_DIGITSLASHWSCOLONCOLONAM, ST_DIGITSLASHWSCOLONCOLONPM: // starts digit then slash 02/ more digits/slashes then whitespace double colons
```

Comment:
I would prefer to have a separate case statement, not bundle it in here and then not do  the strings.Replace ?